internal/core/src/index/StringIndexMarisa.cpp

```diff
@@ -256,18 +257,21 @@ StringIndexMarisa::LoadWithoutAssemble(const BinarySet& set,
     auto len = index->size;
 
     auto file = File::Open(file_name, O_RDWR | O_CREAT | O_EXCL);
-    auto written = file.Write(index->data.get(), len);
-    if (written != len) {
+    try {
+        auto written = file.Write(index->data.get(), len);
+    } catch (const SegcoreError& e) {
         file.Close();
         remove(file_name.c_str());
-        throw SegcoreError(
-            ErrorCode::UnistdError,
-            fmt::format("write index to fd error: {}", strerror(errno)));
+        throw;
```

Comment:
throw e.

Reply:
`throw;` will rethrow the current exception.